### PR TITLE
[DCP][BC] Remove the dependency on _shard.TensorProperties

### PR DESCRIPTION
**Pull request by fegin**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #117773
* #117772
* __->__ #116248

ShardedTensor is in the maintence mode and is going to be deprecated. DCP's metadata should not rely on any definitions in ShardedTensor. This PR creates a replica of TensorProperties in DCP and removes the dependency on _shard.TensorProperties

Differential Revision: [D52357732](https://our.internmc.facebook.com/intern/diff/D52357732/)

cc @mrshenli @pritamdamania87 @zhaojuanmao @satgera @rohan-varma @gqchen @aazzolini @osalpekar @jiayisuse @H-Huang @kwen2501 @awgu @penguinwu @XilunWu @wanchaol @fduwjj @wz337 @tianyu-l @wconstab @yf225 @LucasLLC

### CODE REVIEW

File: torch/distributed/checkpoint/metadata.py [hunk 1]
```diff
@@ -22,6 +24,79 @@ class ChunkStorageMetadata:
...
+    TORCH_CHANNELS_LAST = 1
+    TORCH_PRESERVE_FORMAT = 2
+
+
+@dataclass
+class TensorProperties:
+    """Properties used to create :class:`Tensor`"""
+
+    # Regular tensor fields
+    dtype: torch.dtype = field(default=torch.get_default_dtype())
```
Comment (Skylion007) [inline torch/distributed/checkpoint/metadata.py:38]:
```suggestion
    dtype: torch.dtype = field(default_factory=torch.get_default_dtype)
```

**Review by wconstab — APPROVED** [review]

File: test/distributed/checkpoint/test_compatibility.py [hunk 2]
```diff
@@ -37,6 +39,26 @@ def test_metadata(self) -> None:
                 "The change may break the BC of distributed checkpoint."
             ) from e
 
+    def test_shardedtensor_dependency(self) -> None:
```
Comment (LucasLLC) [inline test/distributed/checkpoint/test_compatibility.py:42]: Nice!

**Review by LucasLLC — APPROVED** [review]
lgtm, great BE effort, thanks!

**Review by wz337 — APPROVED** [review]
LGTM!

File: test/distributed/checkpoint/test_compatibility.py (hunk 2 above)
Comment (LucasLLC) [reply test/distributed/checkpoint/test_compatibility.py:42]: Nit: This should maybe be `test_sharded_tensor_dependency`

**Pull request closed without merging**